Irmaku/update node18

irma-kurnia-alaska opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Resolves: #188 , #189 , #190 , #191 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)


## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

irma-kurnia-phtn committed:
perf(es5): remove legacy es5 bundling resources #188

Changes to be committed:
	modified:   README.md
	deleted:    babel.config.js
	modified:   package-lock.json
	modified:   package.json
	deleted:    rollup.config.js
	new file:   rollup.config.mjs
	deleted:    index.html

irma-kurnia-phtn committed:
perf: update to Lit2.0, TS support, issue templates, component registration extension and update to Node 18+

Change to be committed:
modified: package-lock.json
modified: package.json
modified: README.md
modified: .github/settings.yml
modified: .github/workflows/testPublish.yml
modified: .husky/pre-commit
modified: scripts/staticStyles-template.js
modified: src/auro-dropdown.js
deleted: .github/ISSUE_TEMPLATE/bug_report.md
deleted: .github/ISSUE_TEMPLATE/feature_request.md
deleted: .github/ISSUE_TEMPLATE/general-support.md
deleted: packageScripts/postinstall.js
new file: tsconfig.json
new file: .github/ISSUE_TEMPLATE/bug_report.yml
new file: .github/ISSUE_TEMPLATE/feature_request.yml
new file: .github/ISSUE_TEMPLATE/general-support.yml
new file: packageScripts/postinstall.mjs

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/AlaskaAirlines/auro-dropdown?pullRequest=192) <br/>All committers have signed the CLA.

irma-kurnia-phtn committed:
perf(husky): update .husky/pre-commit and necessary pre-commit actions

Changes to be committed:
modified: .husky/pre-commit
modified: package.json
modified: scripts/postCss.js
modified: src/style.scss
modified: .stylelintrc
deleted: scripts/removeNonRemPlugin.js
deleted: src/style-fixed.scss

irma-kurnia-phtn committed:
docs: update https://unpkg.com into https://cdn.jsdelivr.net/npm

Change to be committed:
modified: demo/index.html

irma-kurnia-phtn committed:
perf: remove unused script

Change to be committed:
modified: package.json

blackfalcon commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,56 @@
+    attributes:
+      label: Please describe the bug
+      description: List out the steps to reproduce the behavior
+      placeholder: 1. Go to '...' 2. Click on '....' 3. Scroll down to '....' 4. See error
```
```suggestion
      description: List out the steps to reproduce the behavior and include screenshots
      placeholder: |
        1. Go to '...'
        2. Click on '....'
        3. Scroll down to '....'
        4. See error
```

blackfalcon commented on package.json:
```diff
@@ -16,71 +16,65 @@
+    "@popperjs/core": "^2.11.7",
+    "chalk": "^5.2.0",
+    "lit": "^2.7.4",
+    "lit-html": "^2.7.3"
```
```suggestion

```

Uninstall this dependency, this is no longer needed with Lit2.0

blackfalcon commented on package.json:
```diff
@@ -16,71 +16,65 @@
+    "@web/dev-server": "^0.2.1",
+    "@web/test-runner": "^0.16.1",
+    "autoprefixer": "^10.4.14",
+    "babel-loader": "^9.1.2",
```
Remove dependency

blackfalcon commented on package.json:
```diff
@@ -16,71 +16,65 @@
     "markdown-magic": "^2.6.1",
-    "marked": "^4.2.5",
-    "nodemon": "^2.0.20",
+    "marked": "^5.0.0",
```
Remove dependency

blackfalcon commented on package.json:
```diff
@@ -16,71 +16,65 @@
+    "postcss": "^8.4.23",
+    "postcss-discard-comments": "^6.0.0",
     "postcss-selector-replace": "^1.0.2",
     "prismjs": "^1.29.0",
```
Remove dependency

blackfalcon commented on package.json:
```diff
@@ -134,37 +128,30 @@
+    "build:sass:demo": "sass ./demo/sass/ --output ./demo/css/",
     "build:sass:component": "sass --no-source-map src:src",
-    "build:watch": "nodemon -e scss,js,html --watch src --watch apiExamples --exec npm run build:dev:assets",
+    "build:watch": "nodemon -e scss,js --watch src --exec npm run build:dev:assets",
```
```suggestion
    "build:watch": "nodemon -e scss,js,html --watch src --watch apiExamples --exec npm run build:dev:assets",
```

Undo this change. The previous script was correct.

blackfalcon commented on package.json:
```diff
@@ -134,37 +128,30 @@
-    "demo:new:build": "mkdir ./build && mkdir ./build/css && mkdir ./build/demo",
-    "demo:rm:build": "rm -rf ./build",
-    "demo:update:index": "node ./scripts/prepForBuild",
+    "cssLint": "stylelint \"./src/*.css\"",
```
Revert

blackfalcon commented on package.json:
```diff
@@ -134,37 +128,30 @@
     "dist:js": "copyfiles -u 1 -V './src/**/*.js' ./dist",
     "esLint": "./node_modules/.bin/eslint src/**/*.js",
-    "linters": "npm-run-all scssLint esLint",
+    "linters": "npm-run-all cssLint esLint",
```
Revert

blackfalcon commented on package.json:
```diff
@@ -134,37 +128,30 @@
   "scripts": {
-    "build": "npm-run-all build:sass sass:render dist:js build:api test build:markdownDocs bundler postinstall",
-    "build:ci": "npm-run-all sweep build",
+    "build": "npm-run-all build:sass sass:render cssLint dist:js bundler postinstall build:api types",
```
```suggestion
    "build": "npm-run-all build:sass sass:render dist:js bundler postinstall build:api types",
```

blackfalcon commented on src/style.scss:
```diff
@@ -3,34 +3,27 @@
-
   background-color: var(--auro-color-background-lightest);
-  box-shadow: -2px 0 5px 2px rgba(0 0 0 / .08), 0 2px 5px 1px rgba(0 0 0 / .08);
+  box-shadow: -2px 0 5px 2px rgb(0 0 0 / .08), 0 2px 5px 1px rgb(0 0 0 / .08);
```
Why was this changed? Please revert.

blackfalcon commented on package.json:
```diff
@@ -134,37 +128,30 @@
     "build:api": "wca analyze 'src/auro-dropdown.js' --outFiles docs/api.md",
-    "build:demo": "npm-run-all build demo:rm:build demo:new:build demo:copy:index demo:copy:demo demo:update:index",
-    "build:dev:assets": "npm-run-all build:sass:demo build:sass:component postCss:component sass:render build:docs",
-    "build:markdownDocs": "node scripts/generateDocs.js",
```
Revert this removed script and update to the following. 

```
"build:markdownDocs": "node scripts/generateDocs.js",
```

blackfalcon commented on .husky/pre-commit:
```diff
@@ -1,4 +1,4 @@
 . "$(dirname "$0")/_/husky.sh"
 
-./node_modules/.bin/npm-run-all preCommit test linters postinstall
+./node_modules/.bin/npm-run-all linters test
```
```suggestion
./node_modules/.bin/npm-run-all preCommit test linters postinstall
```

blackfalcon commented on .stylelintrc:
```diff
@@ -1,137 +1,9 @@
+    "no-duplicate-selectors": null,
+    "font-family-no-duplicate-names": null,
+    "no-descending-specificity": null,
+    "selector-type-no-unknown": null
```
Please revert the whole file. It should be the same as this resource. 

https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/.stylelintrc

irma-kurnia-phtn committed:
perf: update general feedback for update node 18

Change to be committed:
modified: .stylelintrc
modified: package-lock.json
modified: package.json
modified: .github/ISSUE_TEMPLATE/bug_report.yml
modified: .husky/pre-commit
modified: src/style.scss
deleted: scripts/pre-commit.js
new file: scripts/pre-commit.mjs

irma-kurnia-alaska commented:
Hi @blackfalcon, 

I will updated as you mention feedback for this PR. Thank you for your feedback.

irma-kurnia-phtn committed:
perf: update webcorestylesheets from "@alaskaairux/webcorestylesheets": "^3.7.3" into "@aurodesignsystem/webcorestylesheets": "^4.2.0"

Change to be committed:
modified: package-lock.json
modified: package.json

blackfalcon committed:
build: remove unnecessary dir from version control

Changes to be committed:
modified:   .gitignore
deleted:    docTemplates/README.md
modified:   .npmignore
deleted:    scripts/prepForBuild.js

blackfalcon commented:
:tada: This PR is included in version 2.9.8 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@aurodesignsystem/auro-dropdown/v/2.9.8)
- [GitHub release](https://github.com/AlaskaAirlines/auro-dropdown/releases/tag/v2.9.8)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
